service-discovery/kubernetes/src/test/java/io/smallrye/stork/servicediscovery/kubernetes/KubernetesServiceDiscoveryTest.java

```diff
@@ -0,0 +1,167 @@
+package io.smallrye.stork.servicediscovery.kubernetes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.awaitility.Awaitility.await;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.server.mock.EnableKubernetesMockClient;
+import io.smallrye.stork.Service;
+import io.smallrye.stork.ServiceInstance;
+import io.smallrye.stork.Stork;
+import io.smallrye.stork.StorkTestUtils;
+import io.smallrye.stork.test.TestConfigProvider;
+
+@EnableKubernetesMockClient(crud = true)
+public class KubernetesServiceDiscoveryTest {
+
+    KubernetesClient client;
+
+    String k8sMasterUrl;
+
+    @BeforeEach
+    void setUp() {
+        TestConfigProvider.clear();
+        System.setProperty(Config.KUBERNETES_TRUST_CERT_SYSTEM_PROPERTY, "true");
+        k8sMasterUrl = client.getMasterUrl().toString();
+    }
+
+    @Test
+    void shouldGetServiceFromK8sDefaultNamespace() throws InterruptedException {
+
+        TestConfigProvider.addServiceConfig("svc", null, "kubernetes",
+                null, Map.of("k8s-host", k8sMasterUrl));
+        Stork stork = StorkTestUtils.getNewStorkInstance();
+
+        Endpoints endpoint = new EndpointsBuilder()
+                .withNewMetadata().withName("svc").endMetadata()
+                .addToSubsets(new EndpointSubsetBuilder()
+                        .addToAddresses(new EndpointAddressBuilder().withIp("10.96.96.231").build(),
+                                new EndpointAddressBuilder().withIp("10.96.96.232").build(),
```

Comment:
should we also have a test for endpoint without ip but with a hostname?

Reply:
mmm no, I will add it.